pkg/deviceplugin/server.go

```diff
@@ -51,27 +51,30 @@ type devicePluginServer interface {
 
 // server implements devicePluginServer and pluginapi.PluginInterfaceServer interfaces.
 type server struct {
-	devType           string
-	grpcServer        *grpc.Server
-	updatesCh         chan map[string]DeviceInfo
-	devices           map[string]DeviceInfo
-	postAllocate      func(*pluginapi.AllocateResponse) error
-	preStartContainer func(*pluginapi.PreStartContainerRequest) error
-	state             serverState
-	stateMutex        sync.Mutex
+	devType             string
+	grpcServer          *grpc.Server
+	updatesCh           chan map[string]DeviceInfo
+	devices             map[string]DeviceInfo
+	postAllocate        func(*pluginapi.AllocateResponse) error
+	preStartContainer   func(*pluginapi.PreStartContainerRequest) error
+	preferredAllocation func(*pluginapi.PreferredAllocationRequest) (*pluginapi.PreferredAllocationResponse, error)
```

Comment:
These closure declarations have become too lengthy. Let's declare new functional types `postAllocateFunc`, `preStartContainerFunc` and `getPreferredAllocationFunc` in this file and use them instead of `func(...)`.

Reply:
see  c87b0b5

Reply:
could you use the functional types in this file too?